src/projects/TodoList/index.js

```diff
@@ -10,21 +12,43 @@ class TodoList extends React.Component {
     this.pressEnter = this.pressEnter.bind(this)
     this.changeCompleted = this.changeCompleted.bind(this)
     this.confirmEditTask = this.confirmEditTask.bind(this)
+    this.toggleCheckBox = this.toggleCheckBox.bind(this)
+    this.addCategory = this.addCategory.bind(this)
     this.state = {
       todos: [
-        { task: 'Take out the bins', completed: false, isEditing: false, editingTask: '' },
-        { task: 'Put a wash load on', completed: true, isEditing: false, editingTask: '' },
-        { task: 'Learn React', completed: false, isEditing: false, editingTask: '' }
+        { task: 'Take out the bins', completed: false, isEditing: false, editingTask: '', categories: [1,2,3,4] },
+        { task: 'Put a wash load on', completed: true, isEditing: false, editingTask: '', categories: [1,4] },
+        { task: 'Learn React', completed: false, isEditing: false, editingTask: '', categories: [3] }
       ],
-      newItem: ''
+      newItem: '',
+      categories: [
+        { id: 1, category: "Urgent", colour: "#dc3545", text: "white", checked: false },
+        { id: 2, category: "Housework", colour: "#007bff", text: "white", checked: false },
+        { id: 3, category: "Shopping", colour: "#28a745", text: "white", checked: false },
+        { id: 4, category: "Birthday", colour: "#ffc107", text: "black", checked: false }
+      ],
+      newCategory: '',
+      newCategoryRed: '0',
+      newCategoryGreen: '0',
+      newCategoryBlue: '0',
+      newCategoryTextColour: 'white',
+      newCategoryBeingCreated: false
```

Comment:
All these state items could be combined into an object because they're all relating to the same thing:
```js
newCategory: {
  text: '',
  red: '',
  ...
}
```
Keeps it all nice and clean and its obvious that everything inside that object is to do with new category. I know this one might be a pain because you're gonna have to fuck about with everything you're passing this into, but im being picky now 😏

Reply:
ughghghghghghghg :(